server/src/main/java/org/apache/cassandra/sidecar/codecs/SidecarInstanceCodecs.java

```diff
@@ -0,0 +1,65 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.cassandra.sidecar.codecs;
+
+import com.google.inject.Singleton;
+import io.vertx.core.buffer.Buffer;
+import io.vertx.core.eventbus.MessageCodec;
+import org.apache.cassandra.sidecar.common.client.SidecarInstance;
+import org.apache.cassandra.sidecar.common.client.SidecarInstanceImpl;
+
+/**
+ * Codecs for Sidecar instances
+ */
+@Singleton
+public class SidecarInstanceCodecs implements MessageCodec<SidecarInstance, SidecarInstance>
+{
+    @Override
+    public void encodeToWire(Buffer buf, SidecarInstance instance)
+    {
+        buf.appendInt(instance.port());
+        CommonCodecs.STRING.encodeToWire(buf, instance.hostname());
+    }
+
+    @Override
+    public SidecarInstance decodeFromWire(int pos, Buffer buf)
+    {
+        final int port = buf.getInt(pos);
+        pos += 4; // advance 4 bytes after reading int
+        return new SidecarInstanceImpl(CommonCodecs.STRING.decodeFromWire(pos, buf), port);
+    }
```

Comment:
Just declare the string codec in the scope of this class. As commented above, there is no need to have a common codecs group. 

```suggestion
    private static final StringMessageCodec STRING_MESSAGE_CODEC = new StringMessageCodec();

    @Override
    public void encodeToWire(Buffer buf, SidecarInstance instance)
    {
        buf.appendInt(instance.port());
        STRING_MESSAGE_CODEC.encodeToWire(buf, instance.hostname());
    }

    @Override
    public SidecarInstance decodeFromWire(int pos, Buffer buf)
    {
        int port = buf.getInt(pos);
        pos += 4; // advance 4 bytes after reading int
        return new SidecarInstanceImpl(STRING_MESSAGE_CODEC.decodeFromWire(pos, buf), port);
    }
```